src/status_im/protocol/web3/inbox.cljs

```diff
@@ -19,13 +20,16 @@
       {:error (.-message e)})))
 
 (defn- response-handler [error-fn success-fn]
-  (fn [response]
-    (let [{:keys [error result]} (parse-json response)]
-      ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
-      ;; assuming empty error is a success response
-      (if (seq error)
-        (error-fn error)
-        (success-fn result)))))
+  (fn handle-response
+    ([response]
+     (let [{:keys [error result]} (parse-json response)]
+       (handle-response error result)))
+    ([error result]
+     ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
+     ;; assuming empty error is a success response
+     (if-not (empty? error)
```

Comment:
I would still use `(if seq)` here https://github.com/bbatsov/clojure-style-guide#nil-punning

Reply:
@dmitryn I see. But if `js/Error` is being passed to `error`, which is not seqable in ClojureScript, how should it be handled?

Reply:
@foopang good point on `js/Error` is not ISeqable (which makes `seq` and `empty` fail). So we could check for `nil?`. The thing is `add-peer` call is kind of legacy method which doesn't conform to web3-like response and could return `{"error": ""}` as a success result. I think `parse-json` could also transform such response to standard web3 response by removing empty "error" attribut). Does it make sense to you?

Reply:
@foopang i think we should have 2 separate response handlers: one for web3 response (mark-trusted-peer and request-messages) and one legacy responses (add-peer) until we would have add-peer response web3 compliant

Reply:
@foopang current solution with multi-arity fn looks good to me, the only thing i'd change is transforming `{"error": ""}` to `{:result true}` in `parse-json` (to workaround `add-peer` non-standard response).

Reply:
@dmitryn I see what you mean. It's fixed in [this commit](https://github.com/status-im/status-react/pull/2838/commits/e3b45e5ab4c532e3c38edb271f266a00a9b2e1ee) now.